inc/Engine/CriticalPath/RESTDelete.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace WP_Rocket\Engine\CriticalPath;
+
+use WP_REST_Request;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+class RESTDelete implements Subscriber_Interface {
+	const ROUTE_NAMESPACE = 'wp-rocket/v1';
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'rest_api_init' => [ 'register_delete_route' ],
+		];
+	}
+
+	/**
+	 * Register Delete CPCSS route in the WP REST API.
+	 *
+	 * @since  3.6
+	 */
+	public function register_delete_route() {
+		register_rest_route(
+			self::ROUTE_NAMESPACE,
+			'cpcss/post/(?P<id>[\d]+)',
+			[
+				'methods'             => 'DELETE',
+				'callback'            => [ $this, 'delete' ],
+				'permission_callback' => function() {
+					return current_user_can( 'rocket_regenerate_critical_css' );
+				},
+			]
+		);
+	}
+	/**
+	 * Delete Post ID CPCSS file.
+	 *
+	 * @since 3.6
+	 *
+	 * @param  WP_REST_Request $request the WP Rest Request object.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function delete( WP_REST_Request $request ) {
+		$post_id                = $request['id'];
+		$critical_css_file_path = rocket_get_constant( 'WP_ROCKET_CRITICAL_CSS_PATH' ) . get_current_blog_id() . "/posts/post-type-{$post_id}.css";
```

Comment:
```suggestion
		$critical_css_file_path = $this->critical_css_path . "/{$post_type}-{$post_id}.css";
```

Inject the base critical path in the constructor (I'm doing the same on my side)
`$this->critical_css_path = $critical_css_path . get_current_blog_id() . '/posts/';`

You can get the `$post_type` by doing `$post_type = get_post_type( $request['id'] );`

Reply:
@Tabrisrp Should the critical css path be: `posts/post-type-1.css` or `posts/post-1.css` where `post` = the post type?

Reply:
@Tabrisrp and @crystinutzaa Should the critical path be for only posts? How about the CPCSS for pages?